clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -680,6 +790,7 @@ private long maybeUpdate(long now, Node node) {
         final RequestCompletionHandler callback;
         final boolean expectResponse;
         final boolean isInternalMetadataRequest; // used to flag metadata fetches which are triggered internally by NetworkClient
+        final boolean isInternalApiVersionRequest; // used to flag api version fetches which are triggered internally by NetworkClient
```

Comment:
Not sure I am following you here. Below is one of the places where `isInternalMetadataRequest` flag is used.
```
if (req.isInternalMetadataRequest)
                metadataUpdater.handleCompletedMetadataResponse(req.header, now, body);
```
Are you suggesting that I change it to something like.
```
if (req.isInternalRequest && body instanceOf MetadataResponse)
                metadataUpdater.handleCompletedMetadataResponse(req.header, now, (MetadataResponse) body);
```
If so, then we are talking about changing `MetadataResponse` interface. Do we really want to do that?

Do you see any other response being initiated by network client any time soon?